### PR TITLE
refactor: migrate more ops to op2 macro

### PR DESCRIPTION
**Pull request by bartlomieju**
Getting closer...

### CODE REVIEW

File: bench_util/benches/op_baseline.rs [hunk 1]
```diff
@@ -22,18 +22,20 @@ fn setup() -> Vec<Extension> {
   vec![bench_setup::init_ops()]
 }
 
-#[op]
+#[op2(fast)]
 fn op_nop() {}
 
 // TODO(bartlomieju): reenable, currently this op generates a fast function,
 // which is wrong, because i64 is not a compatible type for fast call.
```
Comment (mmastrac) [inline bench_util/benches/op_baseline.rs:29]:
```suggestion
```

**Review by mmastrac — APPROVED** [review]
LGTM one suggestion

File: ext/web/lib.rs [hunk 2]
```diff
@@ -366,32 +364,11 @@ impl Resource for TextDecoderResource {
...
-  ) as u32;
-  out_buf[0] = nchars as u32;
-  Ok(())
-}
-
-#[op(fast, slow = op_encoding_encode_into_fallback)]
+#[op2(fast)]
 fn op_encoding_encode_into(
-  input: Cow<'_, str>,
-  buffer: &mut [u8],
```
Comment (littledivy) [inline ext/web/lib.rs:393]: This is slower for non-latin-1 strings.

**Review by littledivy — CHANGES_REQUESTED** [review]
I am not in favour of removing `op_encoding_encode_into_fallback`.

File: ext/web/lib.rs (hunk 2 above)
Comment (mmastrac) [reply ext/web/lib.rs:393]: Good point. I should implement this

**Review by littledivy — APPROVED** [review]
LGTM

**Pull request merged**